src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -1,186 +1,182 @@
-          display: flex;
-          justify-content: space-between;
-        }
-        .header .name {
-          @apply --paper-font-common-nowrap;
-        }
-        .state-card-dialog {
-          cursor: pointer;
-        }
-      </style>
-    `;
-  }
-
-  private renderEntity(entityConf) {
-    const element = createRowElement(entityConf);
-    if (this._hass) {
-      element.hass = this._hass;
-    }
-    if (
-      entityConf.entity &&
-      !DOMAINS_HIDE_MORE_INFO.includes(computeDomain(entityConf.entity))
-    ) {
-      element.classList.add("state-card-dialog");
-      element.addEventListener("click", () => this._handleClick(entityConf));
-    }
-
-    return element;
-  }
-
-  private _handleClick(entityConf: ConfigEntity) {
-    const entityId = entityConf.entity;
-
-    fireEvent(this, "hass-more-info", { entityId });
-  }
-}
-
-declare global {
-  interface HTMLElementTagNameMap {
-    "hui-entities-card": HuiEntitiesCard;
-  }
-}
-
-customElements.define("hui-entities-card", HuiEntitiesCard);
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+
+import "../../../components/ha-card.js";
+import "../components/hui-entities-toggle.js";
+
+import { fireEvent } from "../../../common/dom/fire_event.js";
+import { DOMAINS_HIDE_MORE_INFO } from "../../../common/const.js";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import createRowElement from "../common/create-row-element.js";
+import computeDomain from "../../../common/entity/compute_domain.js";
+import processConfigEntities from "../common/process-config-entities";
+import { HomeAssistant } from "../../../types.js";
+import { EntityConfig, EntityRow } from "../entity-rows/types.js";
+
+interface ConfigEntity extends EntityConfig {
+  type?: string;
+  secondary_info: "entity-id" | "last-changed";
+  action_name?: string;
+  service?: string;
+  service_data?: object;
+  url?: string;
+}
+
+interface Config extends LovelaceConfig {
+  show_header_toggle?: boolean;
+  title?: string;
+  entities: ConfigEntity[];
+}
+
+class HuiEntitiesCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected _hass?: HomeAssistant;
+  protected _config?: Config;
+  protected _configEntities?: ConfigEntity[];
+
+  set hass(hass) {
+    this._hass = hass;
+    this.shadowRoot!.querySelectorAll("#states > *").forEach(
+      (element: unknown) => {
+        (element as EntityRow).hass = hass;
+      }
+    );
+  }
+
+  static get properties(): PropertyDeclarations {
+    return {
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    if (!this._config) {
+      return 0;
+    }
+    // +1 for the header
+    return (this._config.title ? 1 : 0) + this._config.entities.length;
+  }
+
+  public setConfig(config: Config) {
+    const entities = processConfigEntities(config.entities);
+    for (const entity of entities) {

```

Comment:
We should drop this check. The `createRowElement` function will render an error card when a type is invalid.